build-infra/build-vm.sls

```diff
@@ -206,29 +206,22 @@ github.com:
 {% endif %}
 
 {% for builder in builders_list %}
-{% set config_baseurl = salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':config:repository:baseurl', 'https://github.com/QubesOS/qubes-') %}
+{% set config_baseurl = salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':config:repository:baseurl', 'ssh://github.com/QubesOS/qubes-') %}
 {% set config_repo = salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':config:repository:component', 'release-configs') %}
 {% set config_file = salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':config:file') %}
 {% set keys =  salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':keys', []) %}
 
-{{builder}}-get:
-  git.latest:
-    - name : https://github.com/QubesOS/qubes-builder
-    - target: {{ builder }}
-    - user: user
-
 {{builder}}-check:
-  cmd.run:
-    - name: git verify-tag --raw "$(git describe)" 2>&1 >/dev/null | grep '^\[GNUPG:\] TRUST_FULLY'
-    - cwd: {{ builder }}
+  cmd.script:
+    - source: salt://build-infra/safe-checkout-sha
```

Comment:
There should be also `creates: {{ builder }}`, otherwise it may download (and rollback) already existing clones.

Reply:
I was not aware of that feature, thanks!